src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -0,0 +1,499 @@
+    for (Sequence genSeq : sequenceSet) {
+      ExecutableSequence eseq = new ExecutableSequence(genSeq);
+      eseq.execute(new DummyVisitor(), new DummyCheckGenerator());
+
+      Object generatedObjectValue = null;
+      ExecutionOutcome outcome = eseq.getResult(eseq.sequence.size() - 1);
+      if (outcome instanceof NormalExecution) {
+        generatedObjectValue = ((NormalExecution) outcome).getRuntimeValue();
+      }
+
+      if (generatedObjectValue != null) {
+        sequenceCollection.add(genSeq);
+      }
+    }
+  }
+
+  /**
+   * Get a set of classes that are utilized by the demand-driven input creation process but were not
+   * explicitly specified by the user.
+   *
+   * @return a set of unspecified classes that are automatically included in the demand-driven input
+   *     creation process
+   */
+  public static Set<Class<?>> getUnspecifiedClasses() {
+    return unspecifiedClasses;
+  }
+
+  /**
+   * Checks if the type was specified by the user. If not, logs the class as an unspecified class.
+   *
+   * @param type the type of the object to check for specification
+   */
+  private static void logIfUnspecified(Type type) {
+    String className;
+    if (type.isArray()) {
+      className = ((ArrayType) type).getElementType().getRuntimeClass().getName();
+    } else {
+      className = type.getRuntimeClass().getName();
+    }
+
+    // Add the class to the unspecified classes if it is not specified by the user.
+    if (!SPECIFIED_CLASSES.contains(className)) {
+      unspecifiedClasses.add(type.getRuntimeClass());
+    }
+  }
+
+  /**
+   * Get a set of classes that are utilized by the demand-driven input creation process but were not
+   * explicitly specified by the user. This method additionally filters out classes that are part of
+   * the Java standard library.
+   *
+   * @return A set of unspecified, non-Java classes that are automatically included in the
+   *     demand-driven input creation process.
+   */
+  public static Set<Class<?>> getNonJavaClasses() {
+    Set<Class<?>> nonJavaClasses = new LinkedHashSet<>();
+    for (Class<?> cls : unspecifiedClasses) {
+      if (!startsWithJava(cls.getName()) && !cls.isPrimitive()) {
+        nonJavaClasses.add(cls);
+      }
+    }
+    return nonJavaClasses;
+  }
+
+  /**
+   * Determines whether a class name starts with "java." or represents a Java array.
+   *
+   * @param className the name of the class
+   * @return true if the class name starts with "java." or represents a Java array, false otherwise
+   */
+  public static boolean startsWithJava(String className) {
+    return className.startsWith("java.") || JDK_CLASS_PATTERN.matcher(className).find();
+  }
+
+  /**
+   * Writes the unspecified classes that are automatically used in demand-driven input creation but
+   * were not explicitly specified by the user to the demand-driven logging file.
+   */
+  public static void writeUnspecifiedClassesToLog() {
```

Comment:
This creates new writers on every call, which is inefficient if the method is called multiple times.